src/pykeen/triples/utils.py

```diff
@@ -89,6 +90,70 @@ def load_triples(
     return df.to_numpy()
 
 
+def load_rdf_star_statements(
+    path: Union[str, pathlib.Path, TextIO],
+    max_num_qualifier_pairs: Optional[int] = 0,
+    delimiter: str = ",",
+    encoding: Optional[str] = None,
+    column_remapping: Optional[Sequence[int]] = None,
+) -> np.ndarray:
+    """
+    Load statements from a file, expected a comma-separated h,r,t,qr1,qv1,qr2,qv2,...
+    :param path:
+        The path to the file.
+    :param max_num_qualifier_pairs:
+        If given, trim statements to maximum length of 2 * max_num_qualifier_pairs + 3
+    :param delimiter: The delimiter between the columns in the file
+    :param encoding: The encoding for the file. Defaults to utf-8.
+    :param column_remapping: A remapping if the three columns do not follow the order head-relation-tail.
+        For example, if the order is head-tail-relation, pass ``(0, 2, 1)``
+    :return:
+        A list of statements in the format (h, r, t, *q), where q are qualifiers.
+    """
+    if isinstance(path, (str, pathlib.Path)):
+        path = str(path)
+        for extension, handler in EXTENSION_IMPORTERS.items():
+            if path.endswith(f".{extension}"):
+                return handler(path)
+
+        for prefix, handler in PREFIX_IMPORTERS.items():
+            if path.startswith(f"{prefix}:"):
+                return handler(path[len(f"{prefix}:"):])
+
+    if encoding is None:
+        encoding = "utf-8"
+    max_len = 2 * max_num_qualifier_pairs + 3  # max statement length
+    if column_remapping is not None:
+        if len(column_remapping) > max_len:
+            raise ValueError("remapping must have length not more than the max statement length")
+
+    # TODO find a way to load files w/o knowing max_len in advance
```

Comment:
you can open a file handle then just read the first row, split by the delimiter, then count. then just close the file handle

Reply:
Ah if it was that easy 😢 Statements in the input files are not sorted "longest to shortest" so the longest statement might appear somewhere in the middle of the file, eg
```
h, r, t, qr1, qv1  # the first row does not have padding commas
...
h, r, t, qr1, qv1, qr2, qv2, qr3, qv3
```

so we'd still need to read through the whole file to identify the maximum length